ModelBundle/EventListener/UpdateSiteNodesThemeListener.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace OpenOrchestra\ModelBundle\EventListener;
+
+use Doctrine\ODM\MongoDB\Event\PostFlushEventArgs;
+use Doctrine\ODM\MongoDB\Event\PreUpdateEventArgs;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+use OpenOrchestra\ModelInterface\Model\SiteInterface;
+use OpenOrchestra\ModelInterface\Repository\NodeRepositoryInterface;
+
+/**
+ * Class UpdateSiteNodesThemeListener
+ */
+class UpdateSiteNodesThemeListener
+{
+    protected $nodeClass;
+
+    protected $nodes = array();
+
+    /**
+     * @param string $nodeClass
+     */
+    public function __construct($nodeClass)
+    {
+        $this->nodeClass = $nodeClass;
+    }
+
+    /**
+     * @param PreUpdateEventArgs $event
+     */
+    public function preUpdate(PreUpdateEventArgs $event)
+    {
+        $document = $event->getDocument();
+        if ($document instanceof SiteInterface && $event->hasChangedField("theme")) {
+            $siteTheme = $document->getTheme()->getName();
+
+            /* @var $nodeRepository NodeRepositoryInterface */
+            $nodeRepository = $event->getDocumentManager()->getRepository($this->nodeClass);
```

Comment:
I would rather inject the nodeRepository service (through the container if there is a loop) rather than using the documentManager->getReppository. With the injection you will have the full nodeRepository which is better.